app/javascript/packs/AddNewRestPopup.jsx

```diff
@@ -0,0 +1,151 @@
+import React from 'react';
+import $ from 'jquery';
+
+class AddNewRestPopup extends React.Component {
+
+    constructor() {
```

Comment:
You can (should) dismiss the constructor and use class [arrow functions](https://stackoverflow.com/questions/31362292/how-to-use-arrow-functions-public-class-fields-as-class-methods), that way you won't need `.bind`.

```jsx
class MyComponent extends React.Component {
  state = {
    foo: 'bar',
  }

  updateChange = () {
    ...
  }
}
```

Also, pre-setting `state` with data smells, it's probably not what you wanna do.

Reply:
Updated in relevant files. Thanks!